csrc/device_lower/utils.h

```diff
@@ -143,6 +143,9 @@ bool isCpAsyncBulk(const Expr* expr);
 //! Short-cut for detecting initialization for cpAsync op.
 bool isCpAsyncInit(const Expr* expr);
 
+//! Returns true if the value is a kir::Predicate with PredicateType::ElectSync
+bool isElectSyncPredicate(const Val* v);
```

Comment:
nit: looks like it's only used once. Is it common enough to have it here?

Reply:
Not at the moment. I'll remove it.